### PR TITLE
Fixes 'in' operator precedence issues.

### PR DESCRIPTION
**Pull request by melichior**
As absurd as it sounds, BYOND seems to evaluate logical operators before the 'in' operator. This PR fixes all the instances I could find where the opposite is assumed. As far as I can tell, in all of these cases the 'in' operation should be evaluated first.

If you want proof that this the case, open up an empty project and play around with this code. See what happens when you switch the && with an ||, and place 0 and 1 in l2. Note how logical operators work in DM (Just like Lua, Short Circuited, where "The return value is equal to the last argument to be evaluated." --  DM REF), and that BYOND will not throw an exception when 'in' is used with something other than a list.

```
        var/l1 = list(1,2,3)
        var/l2 = list(3,4,5)
        var/a = 3

        if (a in l1 && a in l2)
            world << "True"
        else
            world << "False"
```

I also found this, which is probably a more credible source than me: http://www.byond.com/forum/?post=1950007

For further proof, look at some of the crazy gibberish my transcompiler spat out on some of these lines.

### CODE REVIEW

**Comment by duncathan** [comment]
wtf

**Comment by Incoming5643** [comment]
Yep, this strange behavior has screwed me over more than a few times

**Comment by RemieRichards** [comment]
Ah yes this is true.
We do ask that things such as this are surrounded in parenthesis, so thank you for correcting existing examples.

File: code/modules/surgery/cavity_implant.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 /datum/surgery_step/handle_cavity/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	if(tool)
-		if(IC || tool.w_class > 3 || NODROP in tool.flags || istype(tool, /obj/item/organ))
+		if(IC || tool.w_class > 3 || (NODROP in tool.flags) || istype(tool, /obj/item/organ))
```
Comment (Incoming5643) [inline code/modules/surgery/cavity_implant.dm]: I think this solves a issue, I forget which one though.

**Comment by tkdrg** [comment]
Looks good to me

**Pull request merged**